### PR TITLE
Add Redis status checks to service health endpoint

### PR DESCRIPTION
**Pull request by bufke**
## Description

Add cache, broker, and session redis checks on service_health endpoint.

## Details

The health endpoint checks postgresql and mongo but not redis. That's inconsistent.

Redis is different in that it's possible to configure multiple redis's for cache, broker, and session store.

### CODE REVIEW

File: kobo/apps/service_health/views.py [hunk 1]
```diff
@@ -37,67 +40,73 @@ def get_response(url_):
...
+def service_health(request):
+    """
+    Return a HTTP 200 if some very basic runtime tests of the application
+    pass. Otherwise, return HTTP 500
+    """
+    all_checks = {
+        'Mongo': lambda: settings.MONGO_DB.instances.find_one(),
+        'Postgres': lambda: Asset.objects.order_by().exists(),
+        'Cache': lambda: cache.set('a', True, 1),
+        'Broker': lambda: celery_app.backend.client.ping(),
```
Comment (jnm) [inline kobo/apps/service_health/views.py:73]: To eliminate the Cypress-based login checker, we need a test for the Redis session store

File: kobo/apps/service_health/views.py [hunk 2]
```diff
@@ -37,67 +40,73 @@ def get_response(url_):
...
+
+    check_results = []
+    any_failure = False
+    for service_name, check_function in all_checks.items():
+        service_message, service_time = check_status(
+            service_name, check_function
+        )
+        any_failure = True if service_message else any_failure
+        check_results.append(
+            f"{service_name}: {service_message or 'OK'} {service_time:.3} seconds"
```
Comment (jnm) [inline kobo/apps/service_health/views.py:87]: Please preserve the "thing *in* x seconds" convention

**Review by jnm — CHANGES_REQUESTED** [review]
Please rename the PR to something that uses the imperative mood, e.g. "Add Redis status checks to service health endpoint"

**Review by jnm — COMMENTED** [review]
@bufke LGTM except the CI failure looks legit?
```
=================================== FAILURES ===================================
__________________ ServiceHealthTestCase.test_service_health ___________________

self = <kobo.apps.service_health.test_service_health.ServiceHealthTestCase testMethod=test_service_health>

    @responses.activate
    def test_service_health(self):
        responses.add(responses.GET, 'http://enketo.mock/', status=200)
        responses.add(
            responses.GET, 'http://kobocat:8001/service_health/', status=200
        )
        res = self.client.get(self.url)
>       self.assertContains(res, "OK")

kobo/apps/service_health/test_service_health.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/site-packages/django/test/testcases.py:462: in assertContains
    text_repr, real_count, msg_prefix = self._assert_contains(
/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/site-packages/django/test/testcases.py:432: in _assert_contains
    self.assertEqual(
E   AssertionError: 500 != 200 : Couldn't retrieve content: Response code was 500 (expected 200)
----------------------------- Captured stderr call -----------------------------
ERROR:django.request:Internal Server Error: /service_health/
------------------------------ Captured log call -------------------------------
ERROR    django.request:log.py:224 Internal Server Error: /service_health/
```

**Comment by bufke** [comment]
I don't know why it passed before, it seems the problem was with the mocked enketo/kc endpoints. Changed the test to use the urls from settings.

**Review by jnm — APPROVED** [review]
Thanks :+1:

**Pull request merged**